src/VisualStudio/Setup/UnifiedSettings/csharpIntellisenseSettings.registration.json

```diff
@@ -0,0 +1,242 @@
+        "text": "@Only_take_effect_when_show_completion_list_after_a_character_is_typed_is_enabled;{13c3bbb4-f18f-4111-9f54-a0fb010d9194}",
+        "visibleWhen": "${config:textEditor.csharp.intellisense.completion.dotnetTriggerCompletionOnTypingLetters}=='false'"
+      }],
+      "migration": {
+        "pass": {
+          "input": {
+            "store": "SettingsManager",
+            "path": "TextEditor.CSharp.Specific.TriggerOnDeletion"
+          }
+        }
+      }
+    },
+    "textEditor.csharp.intellisense.completion.dotnetTriggerCompletionInArgumentLists": {
+      "title": "@Automatically_show_completion_list_in_argument_lists;..\\Microsoft.VisualStudio.LanguageServices.CSharp.dll",
+      "type": "boolean",
+      "default": true,
+      "requiresRestart": false,
+      "order": 10,
+      "migration": {
+        "pass": {
+          "input": {
+            "store": "SettingsManager",
+            "path": "TextEditor.CSharp.Specific.TriggerInArgumentLists"
+          }
+        }
+      }
+    },
+    "textEditor.csharp.intellisense.completion.dotnetHighlightMatchingPortionsOfCompletionListItems": {
+      "title": "@Highlight_matching_portions_of_completion_list_items;..\\Microsoft.VisualStudio.LanguageServices.CSharp.dll",
+      "type": "boolean",
+      "default": true,
+      "requiresRestart": false,
+      "order": 20,
+      "migration": {
+        "pass": {
+          "input": {
+            "store": "SettingsManager",
+            "path": "TextEditor.CSharp.Specific.HighlightMatchingPortionsOfCompletionListItems"
+          }
+        }
+      }
+    },
+    "textEditor.csharp.intellisense.completion.dotnetShowCompletionItemFilters": {
+      "title": "@Show_completion_item_filters;..\\Microsoft.VisualStudio.LanguageServices.CSharp.dll",
+      "type": "boolean",
+      "default": true,
+      "requiresRestart": false,
+      "order": 30,
+      "migration": {
+        "pass": {
+          "input": {
+            "store": "SettingsManager",
+            "path": "TextEditor.CSharp.Specific.ShowCompletionItemFilters"
+          }
+        }
+      }
+    },
+    "textEditor.csharp.intellisense.completion.csharpCompleteStatementOnSemicolon": {
+      "title": "@Automatically_complete_statement_on_semicolon;..\\Microsoft.VisualStudio.LanguageServices.CSharp.dll",
+      "type": "boolean",
+      "default": true,
+      "requiresRestart": false,
+      "order": 40,
+      "migration": {
+        "pass": {
+          "input": {
+            "store": "SettingsManager",
+            "path": "TextEditor.AutomaticallyCompleteStatementOnSemicolon"
+          }
+        }
+      }
+    },
+    "textEditor.csharp.intellisense.completion.dotnetSnippetsBehavior": {
+      "title": "@Automatically_complete_statement_on_semicolon;..\\Microsoft.VisualStudio.LanguageServices.CSharp.dll",
+      "type": "string",
+      "enum": [ "Default", "NeverInclude", "AlwaysInclude", "IncludeAfterTypingIdentifierQuestionTab" ],
```

Comment:
The snippets option on old page:
![image](https://github.com/dotnet/roslyn/assets/24360909/515b2994-559c-41cb-99b6-aa84334ed526)
It uses three radio buttons, but there are four states actually.
By default when nothing is checked it is `default`, when something is clicked user has no way to change to `default` in old UI.

But in new UI, since we don't own it anymore, user could set back to deafult.

![image](https://github.com/dotnet/roslyn/assets/24360909/f24aeb70-12c8-48fb-ab24-0def08529c13)

Reply:
What is 'Default'?

Reply:
I don't know : ) I am waiting @akhera99  to give me good string here

Reply:
Update:
Looks like the `Default` value (for both snippetsRule and enterKeyRule) is a value we used to provide language specfic override.
In https://github.com/dotnet/roslyn/blob/cda55873bfd7f16ac6176f467f98ddf8f644e101/src/Features/Core/Portable/Completion/CompletionService.cs#L81

This is overridden by VB, C#, TS, and F# differently.
(For C# https://github.com/dotnet/roslyn/blob/cda55873bfd7f16ac6176f467f98ddf8f644e101/src/Features/CSharp/Portable/Completion/CSharpCompletionService.cs#L50)
So after a talk with Ankita, it feels like we should show the exact value for C# here. (Because it is the csharp setting page) `Default` value in Enum is still needed because we don't want to break F# and TS folks.

Reply:
To achieve this we have to create something special.
e.g. 'Always Include Snippets'  should be mapped to both '0, and 1'. The rest of the option value would be mapped to 2 & 3 (each enum value maps to one int)
Need to ask if this is could be achieved. Will follow up on next week.

Reply:
Updated: 
`Default` is not removed.
I followed the code here https://github.com/dotnet/roslyn/blob/60f598f20c0919477ed74dcb3e795bb950e566e0/src/Features/CSharp/Portable/Completion/CSharpCompletionService.cs#L57 to specify the default value here.
Will clean this code in another PR

Reply:
![image](https://github.com/dotnet/roslyn/assets/24360909/bb2dd6be-2fe8-4175-a739-d7afedd7b8c5)